File: tags/NHI1-0.17/theLink/example/php/Filter5.php
<?php
#+
#:  \file       theLink/example/php/Filter5.php
#:  \brief      \$Id$
#:  
#:  (C) 2011 - NHI - #1 - Project - Group
#:  
#:  \version    \$Rev$
#:  \author     EMail: aotto1968 at users.berlios.de
#:  \attention  this software has GPL permissions to copy
#:              please contact AUTHORS for additional information
#:

// F1 *******************************************************

class F1 extends MqS implements iServerSetup {
  public function ServerSetup() {
    $this->ServiceCreate('+FTR', array(&$this, 'FTRcmd'));
    $this->ServiceProxy('+EOF');
  }
  public function FTRcmd() {
    $ftr = $this->ServiceGetFilter();
    $ftr->SendSTART();
    $ftr->SendC('F1');
    $ftr->SendC($ftr->ConfigGetName());
    $ftr->SendI($ftr->ConfigGetStartAs());
    $ftr->SendI($this->ConfigGetStatusIs());
    while ($this->ReadItemExists()) {
      $ftr->SendC($this->ReadC());
    }
    $ftr->SendEND_AND_WAIT('+FTR');
    $this->SendRETURN();
  }
}

// F2 *******************************************************

class F2 extends MqS implements iServerSetup {
  public function ServerSetup() {
    $this->ServiceCreate('+FTR', array(&$this, 'FTRcmd'));
    $this->ServiceProxy('+EOF');
  }
  public function FTRcmd() {
    $ftr = $this->ServiceGetFilter();
    $ftr->SendSTART();
    $ftr->SendC('F2');
    $ftr->SendC($ftr->ConfigGetName());
    $ftr->SendI($ftr->ConfigGetStartAs());
    $ftr->SendI($this->ConfigGetStatusIs());
    while ($this->ReadItemExists()) {
      $ftr->SendC($this->ReadC());
    }
    $ftr->SendEND_AND_WAIT('+FTR');
    $this->SendRETURN();
  }
}

// F3 *******************************************************

class F3 extends MqS implements iServerSetup {
  public function ServerSetup() {
    $this->ServiceCreate('+FTR', array(&$this, 'FTRcmd'));
    $this->ServiceProxy('+EOF');
  }
  public function FTRcmd() {
    $ftr = $this->ServiceGetFilter();
    $ftr->SendSTART();
    $ftr->SendC('F3');
    $ftr->SendC($ftr->ConfigGetName());
    $ftr->SendI($ftr->ConfigGetStartAs());
    $ftr->SendI($this->ConfigGetStatusIs());
    while ($this->ReadItemExists()) {
      $ftr->SendC($this->ReadC());
    }
    $ftr->SendEND_AND_WAIT('+FTR');
    $this->SendRETURN();
  }
}

// Main ******************************************************

FactoryAdd('F1');
FactoryAdd('F2');
FactoryAdd('F3');

#fputs(STDERR, "111111111111111 -> " . join(", ", $argv) . "\n");

# first argument (the script-name) is not needed
array_shift($argv);

$srv = FactoryGetCalled($argv[0])->New();
try {
  $srv->LinkCreate($argv);
  $srv->ProcessEvent(MqS::WAIT_FOREVER);
} catch (Exception $ex) {
  $srv->ErrorSet($ex);
}
$srv->Exit();

?>

File: tags/NHI1-0.17/theLink/example/python/Filter5.py
#+
#:  \file       theLink/example/python/Filter5.py
#:  \brief      \$Id$
#:  
#:  (C) 2010 - NHI - #1 - Project - Group
#:  
#:  \version    \$Rev$
#:  \author     EMail: aotto1968 at users.berlios.de
#:  \attention  this software has GPL permissions to copy
#:              please contact AUTHORS for additional information
#:

import sys
from pymsgque import *

# F1 ******************************************************

class F1(MqS):
  def __init__(self, tmpl=None):
    self.ConfigSetServerSetup(self.ServerSetup)
    MqS.__init__(self)
  def ServerSetup(self):
    self.ServiceCreate("+FTR", self.FTRcmd)
    self.ServiceProxy("+EOF")
  def FTRcmd(ctx):
    ftr = ctx.ServiceGetFilter()
    ftr.SendSTART()
    ftr.SendC("F1")
    ftr.SendC(ftr.ConfigGetName())
    ftr.SendI(ftr.ConfigGetStartAs())
    ftr.SendI(ctx.ConfigGetStatusIs())
    while ctx.ReadItemExists():
      ftr.SendC(ctx.ReadC())
    ftr.SendEND_AND_WAIT("+FTR")
    ctx.SendRETURN()

# F2 ******************************************************

class F2(MqS):
  def __init__(self, tmpl=None):
    self.ConfigSetServerSetup(self.ServerSetup)
    MqS.__init__(self)
  def ServerSetup(self):
    self.ServiceCreate("+FTR", self.FTRcmd)
    self.ServiceProxy("+EOF")
  def FTRcmd(ctx):
    ftr = ctx.ServiceGetFilter()
    ftr.SendSTART()
    ftr.SendC("F2")
    ftr.SendC(ftr.ConfigGetName())
    ftr.SendI(ftr.ConfigGetStartAs())
    ftr.SendI(ctx.ConfigGetStatusIs())
    while ctx.ReadItemExists():
      ftr.SendC(ctx.ReadC())
    ftr.SendEND_AND_WAIT("+FTR")
    ctx.SendRETURN()

# F3 ******************************************************

class F3(MqS):
  def __init__(self, tmpl=None):
    self.ConfigSetServerSetup(self.ServerSetup)
    MqS.__init__(self)
  def ServerSetup(self):
    self.ServiceCreate("+FTR", self.FTRcmd)
    self.ServiceProxy("+EOF")
  def FTRcmd(ctx):
    ftr = ctx.ServiceGetFilter()
    ftr.SendSTART()
    ftr.SendC("F3")
    ftr.SendC(ftr.ConfigGetName())
    ftr.SendI(ftr.ConfigGetStartAs())
    ftr.SendI(ctx.ConfigGetStatusIs())
    while ctx.ReadItemExists():
      ftr.SendC(ctx.ReadC())
    ftr.SendEND_AND_WAIT("+FTR")
    ctx.SendRETURN()

# main ****************************************************

FactoryAdd(F1)
FactoryAdd(F2)
FactoryAdd(F3)

srv = FactoryGetCalled(sys.argv[1]).New()
try:
  srv.LinkCreate(sys.argv[1:])
  srv.ProcessEvent(MqS_WAIT_FOREVER)
except:
  srv.ErrorSet()
finally:
  srv.Exit()

